fearless/Modules/Staking/Analytics/AnalyticsValidators/ViewModel/AnalyticsValidatorsViewModelFactory.swift

```diff
@@ -0,0 +1,287 @@
+            .compactMap { validatorId in
+                let validatorAddress = try? addressFactory.addressFromAccountId(
+                    data: validatorId,
+                    type: self.chain.addressType
+                )
+                guard
+                    let address = validatorAddress,
+                    !distinctValidators.contains(address)
+                else { return nil }
+
+                let icon = try? self.iconGenerator.generateFromAddress(address)
+                let validatorName = (identitiesByAddress?[address]?.displayName) ?? address
+                return AnalyticsValidatorItemViewModel(
+                    icon: icon,
+                    validatorName: validatorName,
+                    amount: 0,
+                    progressPercents: 0,
+                    progressText: progressText,
+                    validatorAddress: address
+                )
+            }
+    }
+
+    private func determineListTitle(page: AnalyticsValidatorsPage, locale: Locale) -> String {
+        switch page {
+        case .activity:
+            return R.string.localizable.stakingAnalyticsStakeAllocation(preferredLanguages: locale.rLanguages)
+        case .rewards:
+            return R.string.localizable.stakingRewardsTitle(preferredLanguages: locale.rLanguages)
+        }
+    }
+
+    private func createChartCenterText(
+        page: AnalyticsValidatorsPage,
+        validators: [AnalyticsValidatorItemViewModel],
+        totalEras: Int,
+        locale: Locale
+    ) -> NSAttributedString {
+        switch page {
+        case .activity:
+            let maxDistinctErasCount = validators.map(\.amount).max() ?? 0
+            let activeStakingErasPercents = Double(maxDistinctErasCount) / Double(totalEras)
+            let percentageString = percentFormatter.string(from: activeStakingErasPercents as NSNumber) ?? ""
+
+            return createChartCenterText(
+                firstLine: R.string.localizable
+                    .stakingAnalyticsActiveStaking(preferredLanguages: locale.rLanguages).uppercased(),
+                secondLine: percentageString,
+                thirdLine: String(
+                    format: R.string.localizable.stakingAnalyticsErasRange(
+                        Int(maxDistinctErasCount).description,
+                        totalEras.description,
+                        preferredLanguages: locale.rLanguages
+                    )
+                ),
+                locale: locale
+            )
+        case .rewards:
+            let totalRewards = validators.map(\.amount).reduce(0.0, +)
+            let totalRewardsText = balanceViewModelFactory.amountFromValue(Decimal(totalRewards))
+                .value(for: locale)
+            return createChartCenterText(
+                firstLine: R.string.localizable.stakingAnalyticsReceivedRewards(preferredLanguages: locale.rLanguages),
+                secondLine: totalRewardsText,
+                thirdLine: "100%",
```

Comment:
Looks like we need to determine unpaid rewards to display proper percentage. Todo this we just need to match stash -> (validators, era) -> rewards if we succeed then reward is paid.